sdk/data.go

```diff
@@ -28,7 +28,23 @@ func (i Inventory) SetItem(key string, field string, value interface{}) {
 
 // Event is the data type to represent arbitrary, one-off messages for key
 // activities on a system.
-type Event map[string]interface{}
+type Event struct {
+	Summary  string `json:"summary"`
+	Category string `json:"category,omitempty"`
+}
+
+// NewEvent creates new Event with values passed as arguments.
+func NewEvent(summary, category string) (Event, error) {
+	var ev Event
+	if summary == "" {
+		return ev, fmt.Errorf("summary of the event cannot be empty")
+	}
+	ev = Event{
+		Summary:  summary,
+		Category: category,
```

Comment:
`category` value should be lower case always? Or we don't care about it? Because in the documentation we have examples with only lower case: https://docs.newrelic.com/docs/infrastructure/integrations-sdk/file-specifications/integration-executable-file-specifications#event-data

Reply:
I guess only lowercase. This will be done on marshalling.

Reply:
@smoya What you mean that it will be done on marshalling? That even though the customer set it to uppercase then on marshalling it will be converted to lowercase?

Reply:
I though you meant the key `category` but you said value. I guess categories should be lowercased but this is just an assumption. We must check our internals.

Reply:
There are no rules for category value. So I won't convert it to lowercase.